Buddies/Activities/View Activity/ViewActivityController.swift

```diff
@@ -180,10 +180,9 @@ class ViewActivityController: UIViewController {
 
     // Gets topics from the root topic store:
     func getTopics(from topicIds: [String]) -> [Topic] {
-        let ids = Array(Set(topicIds))
         let appDelegate = UIApplication.shared.delegate as! AppDelegate
-        let topics = appDelegate.topicCollection.topics
-        return ids.map { id in topics.first { $0.id == id } }.compactMap { $0 }
+        let topics = appDelegate.topicCollection.topics.filter { topicIds.contains($0.id) }
```

Comment:
The previous code accounted for the (unlikely) case where the from topicIds contains duplicates. could still handle this by adding line 183 back in

Reply:
Since we `filter` with `contains`, only if `topicCollections` has duplicates will we have duplicates.  So this shouldn't be a problem unless our root problem comes back.